File: Java/SurfaceDemo/surface/src/main/java/com/appbuilders/surface/Tabs/TabBarItem.java
package com.appbuilders.surface.Tabs;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Typeface;
import android.graphics.drawable.Drawable;
import android.view.Gravity;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import com.appbuilders.surface.SfPanel;

/**
 * Created by Erick Sanchez - App Builders CTO
 * Revision 1 - 07/10/17
 */

public class TabBarItem {

    protected View view = null;
    protected View selectedView = null;
    protected SfPanel panel = null;
    protected SfPanel selectedPanel = null;
    protected Class clazz = null;

    public void setView(View view) {

        this.view = view;
    }

    public View getView() {

        return this.view;
    }

    public void setSelectedView(View view) {

        this.selectedView = view;
    }

    public View getSelectedView() {

        return this.selectedView;
    }

    public void setPanel(SfPanel panel) {

        this.panel = panel;
    }

    public SfPanel getPanel() {

        return this.panel;
    }

    public void setSelectedPanel(SfPanel panel) {

        this.selectedPanel = panel;
    }

    public SfPanel getSelectedPanel() {

        return this.selectedPanel;
    }

    public void setClazz(Class clazz) {

        this.clazz = clazz;
    }

    public Class getClazz() {

        return this.clazz;
    }

    public static TabBarItem creteCustomItem(Context context, Class clazz, SfPanel panel, SfPanel selectedPanel) {

        TabBarItem item = new TabBarItem();

        item.setClazz(clazz);
        item.setPanel(panel);
        item.setSelectedPanel(selectedPanel);

        return item;
    }

    public static TabBarItem createBasicItem(Context context, String sequence, Class clazz ,int background, int selectedBackground) {

        Typeface font =  Typeface.createFromAsset( context.getAssets(), "FontAwesome.ttf");
        TabBarItem item = new TabBarItem();

        Button button = new Button(context);
        button.setBackgroundColor(background);
        button.setTypeface(font);
        button.setText(sequence);
        button.setTextSize((float) (12 * context.getResources().getDisplayMetrics().density));
        //button.setGravity(Gravity.TOP | Gravity.CENTER_HORIZONTAL);
        //button.setGravity(Gravity.TOP | Gravity.CENTER_HORIZONTAL);

        Button buttonSelected = new Button(context);
        buttonSelected.setBackgroundColor(selectedBackground);
        //buttonSelected.setBackgroundResource(R.drawable.tags_rounded_corners);
        //GradientDrawable drawable = (GradientDrawable) buttonSelected.getBackground();
        //drawable.setColor(backgroundSelectd);
        buttonSelected.setTypeface(font);
        buttonSelected.setText(sequence);
        buttonSelected.setTextSize((float) (12 * context.getResources().getDisplayMetrics().density));
        //buttonSelected.setGravity(Gravity.TOP | Gravity.CENTER_HORIZONTAL);

        item.setView(button);
        item.setSelectedView(buttonSelected);
        item.setClazz(clazz);

        return item;
    }

    public static TabBarItem createImageItem(Context context, Class clazz, int imageId, int imageSelectedId) {

        TabBarItem item = new TabBarItem();

        ImageView image = new ImageView(context);
        Bitmap bitmap = BitmapFactory.decodeStream(context.getResources().openRawResource(imageId));
        image.setImageBitmap(bitmap);

        ImageView imageSelected = new ImageView(context);
        Bitmap bitmapSelected = BitmapFactory.decodeStream(context.getResources().openRawResource(imageSelectedId));
        imageSelected.setImageBitmap(bitmapSelected);

        item.setView(image);
        item.setSelectedView(imageSelected);
        item.setClazz(clazz);

        return item;
    }

}

File: Swift/SurfaceEngineDemo/SurfaceEngineDemo/SurfaceEngine/Controls/Tabs/TabBarItem.swift
//
//  TabBarItem.swift
//  SurfaceEngineDemo
//
//  Created by Erick Sanchez on 13/10/17.
//  Copyright © 2017 App Builders. All rights reserved.
//

import UIKit

public class TabBarItem {
    
    internal var view:UIView!;
    internal var selectedView:UIView!;
    internal var panel:SfPanel?;
    internal var selectedPanel:SfPanel!;
    internal var clazz:UIViewController!;
    
    public init() {
        
        self.view = nil;
        self.selectedView = nil;
        self.panel = nil;
        self.selectedPanel = nil;
        self.clazz = UIViewController();
    }
    
    public func setView(view:UIView) {
        self.view = view;
    }
    
    public func getView() -> UIView {
        return self.view!;
    }
    
    public func setSelectedView(view:UIView) {
        self.selectedView = view;
    }
    
    public func getSelectedView() -> UIView {
        return self.selectedView!;
    }
    
    public func setPanel(panel:SfPanel) {
        self.panel = panel;
    }
    
    public func getPanel() -> SfPanel? {
        return self.panel;
    }
    
    public func setSelectedPanel(panel:SfPanel) {
        self.selectedPanel = panel;
    }
    
    public func getSelectedPanel() -> SfPanel {
        return self.selectedPanel!;
    }
    
    public func setClazz(clazz:UIViewController) {
        self.clazz = clazz;
    }
    
    public func getClazz() -> UIViewController {
        return self.clazz;
    }
    
    public static func createCustomItem(clazz:UIViewController, panel:SfPanel, selectedPanel:SfPanel) -> TabBarItem {
    
        let item = TabBarItem();
        item.setClazz(clazz: clazz);
        item.setPanel(panel: panel);
        item.setSelectedPanel(panel: selectedPanel);
        return item;
    }
    
    public static func createBasicItem(clazz:UIViewController, sequence:String, background:UIColor, backgroundSelected:UIColor) -> TabBarItem {
        
        let item = TabBarItem();
        
        let button = UIButton();
        button.backgroundColor = background;
        button.setTitle(sequence, for: UIControlState.normal);
        button.setTitleColor(UIColor.black, for: UIControlState.normal);
        
        let buttonSelected = UIButton();
        buttonSelected.backgroundColor = backgroundSelected;
        buttonSelected.setTitle(sequence, for: UIControlState.normal);
        
        item.setView(view: button);
        item.setSelectedView(view: buttonSelected);
        item.setClazz(clazz: clazz);
        
        return item;
    }
    
    public static func createImageItem(clazz:UIViewController, imageId:UIImage, imageSelectedId:UIImage) -> TabBarItem {
        
        let item = TabBarItem();
        
        let image = UIImageView(image: imageId);
        let imageSelected = UIImageView(image: imageSelectedId);
        
        item.setView(view: image);
        item.setSelectedView(view: imageSelected);
        item.setClazz(clazz: clazz);
        return item;
    }
}
